framework/resource/retryresource/crud_resource_wrapper.go

```diff
@@ -0,0 +1,100 @@
+package retryresource
+
+import (
+	"context"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	originalframework "github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/internal"
+	"github.com/giantswarm/operatorkit/framework/resource/internal/framework"
+)
+
+// crudResourceWrapper is a specialized wrapper which wraps
+// *framework.CRUDResource.
+type crudResourceWrapper struct {
+	logger   micrologger.Logger
+	resource framework.Resource
+
+	backOff backoff.BackOff
+}
+
+func newCRUDResourceWrapper(config Config) (*crudResourceWrapper, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.Resource == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Resource must not be empty")
+	}
+
+	if config.BackOff == nil {
+		config.BackOff = backoff.NewExponentialBackOff()
+	}
+
+	// Wrap underlying resource Ops with retry logic. Underlying resource
+	// is a pointer so we can modify it in place.
+	{
+		underlying, ok := internal.Underlying(config.Resource).(*originalframework.CRUDResource)
+		if !ok {
+			return nil, microerror.Maskf(incompatibleUnderlyingResourceError, "expected %T", underlying)
+		}
+
+		c := crudResourceOpsWrapperConfig{
+			Logger: config.Logger,
+			Ops:    underlying.CRUDResourceOps,
+
+			BackOff: config.BackOff,
+		}
+
+		wrappedOps, err := newCRUDResourceWrapperOps(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		underlying.CRUDResourceOps = wrappedOps
```

Comment:
This isn't beautiful. I couldn't find better idea than to swap ops with wrapped one. We can do a full copy of a resource. At this point I don't think it's that important. The interface for the end user won't change.